src/Views/AccountView.vala

```diff
@@ -265,6 +284,137 @@ public class Installer.AccountView : AbstractInstallerView {
+        Utils.set_hostname (hostname_entry.text);
+    }
+
+    private async void set_locale () {
+        string lang = Configuration.get_default ().lang;
+        string? locale = null;
+        bool success = yield LocaleHelper.language2locale (lang, out locale);
+
+        if (!success || locale == null || locale == "") {
+            warning ("Falling back to setting unconverted language as user's locale, may result in incorrect language");
+            created_user.set_language (lang);
+        } else {
+            created_user.set_language (locale);
+        }
+    }
+
+    private async void set_accounts_service_settings () {
+        AccountsService accounts_service = null;
+
+        try {
+            var act_service = yield GLib.Bus.get_proxy<FDO.Accounts> (GLib.BusType.SYSTEM,
+                                                                      "org.freedesktop.Accounts",
+                                                                      "/org/freedesktop/Accounts");
+            var user_path = act_service.find_user_by_name (created_user.user_name);
+
+            accounts_service = yield GLib.Bus.get_proxy (GLib.BusType.SYSTEM,
+                                                        "org.freedesktop.Accounts",
+                                                        user_path,
+                                                        GLib.DBusProxyFlags.GET_INVALIDATED_PROPERTIES);
+        } catch (Error e) {
+            warning ("Unable to get AccountsService proxy, settings on new user may be incorrect: %s", e.message);
+        }
+
+        if (accounts_service != null) {
+            var layouts = Configuration.get_default ().keyboard_layout.to_accountsservice_array ();
+            if (Configuration.get_default ().keyboard_variant != null) {
+                layouts = Configuration.get_default ().keyboard_variant.to_accountsservice_array ();
+            }
+
+            accounts_service.keyboard_layouts = layouts;
+            accounts_service.left_handed = Configuration.get_default ().left_handed;
+        }
+    }
+
+    private bool is_taken_username (string username) {
+        foreach (unowned Act.User user in user_manager.list_users ()) {
+            if (user.get_user_name () == username) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private bool is_valid_username (string username) {
+        try {
+            if (new Regex ("^[a-z]+[a-z0-9]*$").match (username)) {
+                return true;
+            }
+            return false;
+        } catch (Error e) {
+            critical (e.message);
+            return false;
+        }
+    }
+
+    private string gen_username (string fullname) {
+        string username = "";
+        bool met_alpha = false;
+
+        foreach (char c in fullname.to_ascii ().to_utf8 ()) {
+            if (c.isalpha ()) {
+                username += c.to_string ().down ();
+                met_alpha = true;
```

Comment:
I just want to note that this means that usernames cannot start with numbers. Is that by design? The warning message, at least in the user management in system settings, if you have a digit first says that the 

> username can only be lowercase letters or numbers

Improper message or bug?

Reply:
So this is just for the auto-generated suggested username when you type in your real full name. I think it's probably okay to assume that a persons real full name doesn't include numbers. They can always edit the suggested username to include numbers if they like

Reply:
Ah I see, I think that is a safe assumption as well. The question came around specifically with _starting_ a username with a number fails though. Dani1337 works but 2Dani2Furious will fail. Pretty minor and not really relevant to the changes this MR is for but thought I would ask.

Reply:
Oh I see. Yeah we should probably handle that. Can you file an issue report?